pkg/controller/usersignup/usersignup_controller_test.go

```diff
@@ -1332,6 +1423,212 @@ func TestUserSignupDeactivatedAfterMURCreated(t *testing.T) {
+			},
+			v1alpha1.Condition{
+				Type:   v1alpha1.UserSignupComplete,
+				Status: v1.ConditionTrue,
+			},
+			v1alpha1.Condition{
+				Type:    v1alpha1.UserSignupUserDeactivatedNotificationCreated,
+				Status:  v1.ConditionFalse,
+				Reason:  "NotificationCRCreationFailed",
+				Message: "unable to create deactivation notification",
+			})
+		AssertThatCounterHas(t, 2)
+
+		// A deactivated notification should have been created
+		notification := &v1alpha1.Notification{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Status.CompliantUsername + "-deactivated", Namespace: userSignup.Namespace}, notification)
+		require.Error(t, err)
+		require.Equal(t, "notifications.toolchain.dev.openshift.com \"john-doe-deactivated\" not found", err.Error())
+	})
+}
+
+func TestUserSignupReactivateAfterDeactivated(t *testing.T) {
+	// given
+	userSignup := &v1alpha1.UserSignup{
+		ObjectMeta: newObjectMeta("", "john.doe@redhat.com"),
+		Spec: v1alpha1.UserSignupSpec{
+			Username:    "john.doe@redhat.com",
+			Deactivated: false,
+		},
+		Status: v1alpha1.UserSignupStatus{
+			Conditions: []v1alpha1.Condition{
+				{
+					Type:   v1alpha1.UserSignupComplete,
+					Status: v1.ConditionTrue,
+				},
+				{
+					Type:   v1alpha1.UserSignupApproved,
+					Status: v1.ConditionTrue,
+					Reason: "ApprovedAutomatically",
+				},
+				{
+					Type:   v1alpha1.UserSignupUserDeactivatedNotificationCreated,
+					Status: v1.ConditionTrue,
+					Reason: "NotificationCRCreated",
+				},
+			},
+			CompliantUsername: "john-doe",
+		},
+	}
+	userSignup.Labels["toolchain.dev.openshift.com/approved"] = "true"
+	key := test.NamespacedName(operatorNamespace, userSignup.Name)
+
+	t.Run("when reactivating the usersignup successfully", func(t *testing.T) {
+		// given
+		InitializeCounter(t, 2)
+		defer counter.Reset()
+		r, req, _ := prepareReconcile(t, userSignup.Name, userSignup, configMap(configuration.UserApprovalPolicyAutomatic), basicNSTemplateTier)
+		createMemberCluster(t, r.client, "member1", ready)
+		defer clearMemberClusters(r.client)
+
+		// when
+		_, err := r.Reconcile(req)
+
+		// then
+		require.NoError(t, err)
+
+		// Lookup the UserSignup
+		err = r.client.Get(context.TODO(), key, userSignup)
+		require.NoError(t, err)
+
+		// Confirm the status has been set to Deactivated
+		test.AssertConditionsMatch(t, userSignup.Status.Conditions,
+			v1alpha1.Condition{
+				Type:   v1alpha1.UserSignupApproved,
+				Status: v1.ConditionTrue,
+				Reason: "ApprovedAutomatically",
+			},
+			v1alpha1.Condition{
+				Type:   v1alpha1.UserSignupComplete,
+				Status: v1.ConditionTrue,
+			},
+			v1alpha1.Condition{
+				Type:   v1alpha1.UserSignupUserDeactivatedNotificationCreated,
+				Status: v1.ConditionFalse,
+				Reason: "NotificationCRNotCreated",
+			})
+		AssertThatCounterHas(t, 3)
+
+		// There should not be a notification created because the user was reactivated
+		notifications := &v1alpha1.NotificationList{}
```

Comment:
Same here. It doesn't seem to be testing what you state. We do need to test the reactivation flow but it seems to be missing.

Reply:
Same as above comment